### PR TITLE
Single GPU testing

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Adds single gpu testing (https://github.com/PyTorchLightning/metrics/issues/35).
If metrics are tested using the `MetricTester` class, and `ddp=False` they will as default now be tested on gpu. If gpu is not available will fallback to cpu testing. 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `__getitem__` as metric arithmetic operation ([#142](https://github.com/PyTorchLightning/metrics/pull/142))
 
 
+- Added testing on single gpu for all metrics (())
```
Comment (SkafteNicki) [inline CHANGELOG.md:50]:
```suggestion
- Added testing on single gpu for all metrics ([#150](https://github.com/PyTorchLightning/metrics/pull/150))
```

File: tests/helpers/testers.py [hunk 2]
```diff
@@ -257,13 +274,19 @@ def run_functional_metric_test(
             kwargs_update: Additional keyword arguments that will be passed with preds and
                 target when running update on the metric.
         """
+        if torch.cuda.is_available():
+            device = 'cuda'
+        else:
+            device = 'cpu'
```
Comment (Borda) [inline tests/helpers/testers.py:280]:
```suggestion
        device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

File: tests/helpers/testers.py [hunk 3]
```diff
@@ -322,6 +345,11 @@ def run_class_metric_test(
                 [(rank, self.poolSize) for rank in range(self.poolSize)],
             )
         else:
+            if torch.cuda.is_available():
+                device = 'cuda'
+            else:
+                device = 'cpu'
```
Comment (Borda) [inline tests/helpers/testers.py:351]:
```suggestion
            device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -47,6 +47,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `__getitem__` as metric arithmetic operation ([#142](https://github.com/PyTorchLightning/metrics/pull/142))
 
 
+- Added testing on single gpu for all metrics ([#150](https://github.com/PyTorchLightning/metrics/pull/150))
```
Comment (Borda) [inline CHANGELOG.md:50]: probably no need for this in chnagelog

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -47,6 +47,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+
```
Comment (Borda) [inline CHANGELOG.md:52]:
```suggestion
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by Borda** [comment]
@SkafteNicki there seems some problem as it expects to have cuda? even running on CPU...

**Pull request merged**